solutions/src/main/java/ru/sberbank/school/task02/ExtendedFxConversionServiceImpl.java

```diff
@@ -1,31 +1,126 @@
 package ru.sberbank.school.task02;
 
+import lombok.NonNull;
 import ru.sberbank.school.task02.util.Beneficiary;
 import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
 import ru.sberbank.school.task02.util.Symbol;
 
 import java.math.BigDecimal;
-import java.util.Optional;
+import java.math.RoundingMode;
+import java.util.*;
 
 /**
  * Created by Mart
  * 10.05.2019
  **/
-public class ExtendedFxConversionServiceImpl implements ExtendedFxConversionService {
-    @Override
-    public Optional<BigDecimal> convertReversed(ClientOperation operation, Symbol symbol,
-                                                BigDecimal amount, Beneficiary beneficiary) {
-        return Optional.empty();
+public class ExtendedFxConversionServiceImpl extends FxConversionServiceImpl implements ExtendedFxConversionService {
+
+    public ExtendedFxConversionServiceImpl(ExternalQuotesService externalQuotesService) {
+        super(externalQuotesService);
     }
 
     @Override
-    public Optional<BigDecimal> convertReversed(ClientOperation operation, Symbol symbol,
-                                                BigDecimal amount, double delta, Beneficiary beneficiary) {
-        return Optional.empty();
+    public Optional<BigDecimal> convertReversed(ClientOperation operation,
+                                                Symbol symbol,
+                                                BigDecimal amount,
+                                                Beneficiary beneficiary) {
+
+        return convertReversed(operation, symbol, amount,0, beneficiary);
     }
 
+    /**
```

Comment:
Summary javadoc is missing.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1461611670>*